### PR TITLE
Fix #4374 menuButton: Add support for submenus

### LINKED ISSUE #4374: menuButton: add support for submenus
PrimeFaces' `<p:menubar>` component supports nested `<p:submenu>`s to group menu-items into sub-menus. This grouping is currently not supported for `<p:menuButton>`. `<p:submenu>`s inside a menuButton is silently ignored. It would be nice to also have sub-menu support in menu-buttons.

## 1) Environment

PrimeFaces 6.2

### PR DESCRIPTION
**Pull request by maruTA-bis5**
This PR add submenus support to menuButton component. Fixes #4374

Example (source: https://github.com/maruTA-bis5/primefaces-test/tree/menubutton-submenu)
image

### CODE REVIEW

**Comment by melloware** [comment]
I am not sure how others will feel extending TieredMenu but I like this change.

File: src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js [hunk 1]
```diff
@@ -54,6 +75,14 @@ PrimeFaces.widget.MenuButton = PrimeFaces.widget.BaseWidget.extend({
      * @private
      */
     bindEvents: function() {
+        this.menuId = this.jqId + '_menu';
+        this.button = this.jq.children('button');
+        this.menu = this.jq.children('.ui-menu');
+        this.menuitems = this.jq.find('.ui-menuitem');
+        this.cfg.disabled = this.button.is(':disabled');
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js]: Not sure why you moved this out of the `init` method?  Usually initialization of values like this is done in the `init` method.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by tandraschko** [comment]
in general it looks ok, also i wonder about mellowares statement
and please dont do things like
        var pos = null;
and asign `pos` in the next statement; just do `var pos = {}`

File: src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js (hunk 1 above)
Comment (maruTA-bis5) [reply src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js]: If initialization is in `init` method, `bindEvents` called from `TieredMenu#init` before initialization and raised script error 
caused by undefined reference.

But I found another solution. I'll fix this.
I'm sorry for my lack of experience with JavaScript.

File: src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js [hunk 2]
```diff
@@ -35,11 +35,37 @@ PrimeFaces.widget.MenuButton = PrimeFaces.widget.BaseWidget.extend({
...
+
+        this.timeoutId = setTimeout(function () {
+           submenu.css('z-index', PrimeFaces.nextZindex())
+                  .show()
+                  .position(pos)
+        }, this.cfg.delay);
+    },
+
     /**
      * @override
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js:70]: You can remove the @override and @inheritdoc here as now that you have renamed the method its no longer overriding the parent.

**Review by melloware — CHANGES_REQUESTED** [review]

File: src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js (hunk 2 above)
Comment (maruTA-bis5) [reply src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js:70]: `bindButtonEvents` method does not have `@override` and `@inheritdoc`. This line is for `refresh` method.

Comment (melloware) [reply src/main/resources/META-INF/resources/primefaces/menu/menu.menubutton.js:70]: sorry the way the review looks the code folding confused me. I see it in the raw file now.

**Review by melloware — APPROVED** [review]

**Comment by tandraschko** [comment]
I will merge it now, would be great if someone coud retest in the showcase

**Comment by tandraschko** [comment]
Thanks for the PR!

**Comment by tandraschko** [comment]
the inhertiance changes breaks jsdoc generation. Can someone check it?

**Comment by melloware** [comment]
I can check this.

**Comment by melloware** [comment]
OK I fixed the JSDocs and I just tested MenuButton in the Showcase and everything is looking good.

**Pull request merged**